src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java

```diff
@@ -39,64 +44,75 @@
+    private final AtomicReference<ScheduledFuture<?>> scheduledSyncFuture = new AtomicReference<>(null);
+    private final ScheduledExecutorService ses;
     private final ChildChanged handleConfigurationChangeLambda = (whatHappened, node) -> {
         if (whatHappened == WhatHappened.timestampUpdated || whatHappened == WhatHappened.interiorAdded) {
             return;
         }
-        if (whatHappened == WhatHappened.initialized || SECRETS_TOPIC.equals(node.getName())) {
+        if (whatHappened == WhatHappened.initialized || SECRETS_TOPIC.equals(node.getName())
+                || PERIODIC_REFRESH_INTERVAL_MIN.equals(node.getName())) {
             serviceChanged();
         }
     };
-
     @Inject
     SecretManagerIPCAgent secretManagerIPCAgent;
-
     @Inject
     private GreengrassCoreIPCService greengrassCoreIPCService;
+    // for testing
+    @Setter
+    private CountDownLatch isInitialSyncComplete = new CountDownLatch(1);
 
     /**
      * Constructor for SecretManagerService Service.
-     * @param topics                root Configuration topic for this service
-     * @param secretManager         secret manager which manages secrets
-     * @param authorizationHandler  authorization handler
-     * @param executorService       executor service
+     *
+     * @param topics               root Configuration topic for this service
+     * @param secretManager        secret manager which manages secrets
+     * @param authorizationHandler authorization handler
+     * @param executorService executor service
+     * @param ses scheduled executor service
      */
     @Inject
-    public SecretManagerService(Topics topics,
-                                SecretManager secretManager,
-                                AuthorizationHandler authorizationHandler,
-                                ExecutorService executorService) {
+    public SecretManagerService(Topics topics, SecretManager secretManager, AuthorizationHandler authorizationHandler,
+                                ExecutorService executorService, ScheduledExecutorService ses) {
         super(topics);
         this.secretManager = secretManager;
         this.authorizationHandler = authorizationHandler;
         this.executor = executorService;
+        this.ses = ses;
     }
 
     @Override
     protected void install() throws InterruptedException {
         super.install();
+        // Re-initialize the sync counter every time the component is installed
+        setIsInitialSyncComplete(new CountDownLatch(1));
         // subscribe will invoke serviceChanged right away to sync from cloud
         this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this.handleConfigurationChangeLambda);
     }
 
     private void serviceChanged() {
-        replaceSyncFuture(secretManager::syncFromCloud);
+        Runnable runnable = () -> secretManager.syncFromCloud(isInitialSyncComplete);
```

Comment:
nit: `SercetManager` shouldn't know about `isInitialSyncComplete` since it's an implementation detail of SecretManagerService

```
scheduleNextSync(() -> {
  try {
   syncFromCloud();
    isInitialSyncComplete.countdown();
 } catch (exception ) {
   // log
 }
})
```